system-addon/lib/TopSitesFeed.jsm

```diff
@@ -137,8 +136,12 @@ this.TopSitesFeed = class TopSitesFeed {
       data: this._getPinnedWithData()
     }));
   }
-  onAction(action) {
+  async onAction(action) {
     switch (action.type) {
+      case at.INIT:
```

Comment:
This could still fail due to a race condition, because `NEW_TAB_LOAD` could be fired before `this._tippyTopProvider` is done initializing. I think it might be better to do something like this inside `refresh`:

```js
if (!this._tippyTopProvider.initialized) await this._tippyTopProvider.init()
// continue refreshing
```